src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
+                is_hit = true;
+                break;
+            }
+
+            if(i > 0){
+                center = line_segments[((actor->mExecuteCount & 0b11) + (i * 4)) % 20];
+                center.y -= 200.0f;
+                actor->mCylArr[i].SetC(center);   
+            }
+        }
+    }
+    else {
+        center.set(0.0f, -20000.0f, 0.0f);
+        actor->mSph.SetC(center);
+        for(int i = 0; i < 5; i++){
+            actor->mCylArr[i].SetC(center);
+        }
+    }
+
+    dComIfG_Ccsp()->Set(&actor->mSph);
+    for(int i = 0; i < 5; i++){
+        dComIfG_Ccsp()->Set(&actor->mCylArr[i]);
+    }
+    
+    if(is_hit || (actor->field_2C4 == 0 && actor->mSph.ChkTgHit() != 0)){
+        if(!is_hit){
+            hit_atInfo.mpObj = actor->mSph.GetTgHitObj();
+            hit_atInfo.pParticlePos = actor->mSph.GetTgHitPosP();
+        }
+        hit_atInfo.mpActor = at_power_check(&hit_atInfo);
+        actor->field_2C4 = 10;
+        if(hit_atInfo.mpActor != NULL){
+            def_se_set(actor, hit_atInfo.mpObj, 33);
+            if(actor->field_30C != NULL){
+                actor->mState = 1;
+                actor->field_2BA = 0;
+                actor->field_2BC[0] = l_HIO.u8;
+                actor->field_2BC[1] = l_HIO.u8 + 90 + REG0_S(2);
+                actor->field_2F8 = 3.0f;
+                actor->field_2FC = 40.0f;
+                actor->field_300 = cM_rndF(20.0f) + 30.0f;
+                *actor->field_314 = 1;
+                cXyz particle_scale(0.5f, 0.5f, 0.5f);
+                dComIfGp_particle_set(19, &hit_atInfo.mpActor->eyePos, NULL, &particle_scale);
+                dComIfGp_particle_set(22, &hit_atInfo.mpActor->eyePos, NULL, &particle_scale);
+            }
+        }
+    }
 }
 
 /* 000021EC-0000225C       .text daShand_Execute__FP11shand_class */
-static BOOL daShand_Execute(shand_class*) {
-    /* Nonmatching */
+static BOOL daShand_Execute(shand_class* actor) {
+    actor->mExecuteCount++;
+    
+    
+    for(int i = 4, j = 0; i != 0; i--, j++){
+        if(actor->field_2BC[j] != 0)
+            actor->field_2BC[j]--;    
+    }
+
+    if(actor->field_2C4 != 0)
+        actor->field_2C4--;
+
+    hand_move(actor);
+
+    return TRUE;
 }
 
 /* 0000225C-00002264       .text daShand_IsDelete__FP11shand_class */
 static BOOL daShand_IsDelete(shand_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00002264-000022D4       .text daShand_Delete__FP11shand_class */
-static BOOL daShand_Delete(shand_class*) {
-    /* Nonmatching */
+static BOOL daShand_Delete(shand_class* actor) {
+    dComIfG_resDelete(&actor->mPhs, "Shand");
+    if(actor->mHasHIO){
+        hio_set = false;
+        mDoHIO_deleteChild(l_HIO.mId);
+    }
+
+    return TRUE;
 }
 
 /* 000022D4-00002360       .text useHeapInit__FP11shand_class */
-void useHeapInit(shand_class*) {
-    /* Nonmatching */
+static BOOL useHeapInit(shand_class* actor) {
+    const int res_index = ((fopAcM_GetParam(actor) & 0xff) == 53) ? 4 : 3;
```

Comment:
You can include d/res/res_shand.h to get an enum with the filenames corresponding to the 4/3 indexes here to make it more readable